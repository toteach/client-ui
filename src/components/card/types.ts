export interface Card {
  isHeaderShown: boolean;
  isImageShown: boolean;
}
