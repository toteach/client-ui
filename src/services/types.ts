export type ValidationError = string;

export interface CommonValidationErrorCollection {
  [key: string]: ValidationError;
}

export interface Error {
  message?: string;
  code?: number;
  status_code?: number;
  validation?: CommonValidationErrorCollection;
}

export interface ServiceResponse<F, FT> {
  original: F;
  transformed: FT;
}
