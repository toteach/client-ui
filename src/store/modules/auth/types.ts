export interface State {
  authenticated: boolean;
  errors: AuthError[];
}

export interface AuthError {
  title: string;
  message: string;
}
