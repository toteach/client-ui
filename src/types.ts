export type TColorModifiers =
    'white' |
    'black' |
    'light' |
    'dark' |
    'primary' |
    'info' |
    'success' |
    'warning' |
    'danger' |
    'text';

export type TSizes =
    'small' |
    'medium' |
    'large';

export type TPositions =
    'top' |
    'left' |
    'bottom' |
    'right';

export type TStates =
    'active' |
    'focus' |
    'hover';


