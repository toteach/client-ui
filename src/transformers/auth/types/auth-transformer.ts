export interface AuthRequestData {
  email: string;
  password: string;
  remember: boolean;
}

export interface AuthResponseData {
  status: number;
  errors?: string[];
}
