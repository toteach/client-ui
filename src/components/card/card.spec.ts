import { shallowMount, Wrapper } from '@vue/test-utils';
import Vue from 'vue';

import TCard from './card.vue';
import { Card as ICard } from './types';

// eslint-disable-next-line max-lines-per-function
describe('TCard', () => {
  let wrapper: Wrapper<Vue>;
  beforeEach(() => {
    wrapper = shallowMount(TCard);
  });

  it('is initialized', () => {
    expect(wrapper.name()).toBe('TCard');
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('title', () => {
    beforeEach(() => {
      const title = 'Title';
      wrapper.setProps({
        title,
      });
    });

    it('has title', () => {
      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
      const card = wrapper.find(TCard).vm as unknown as ICard;
      expect(card.isHeaderShown).toBeTruthy();
    });
    it('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('image', () => {
    const image = 'https://www.fillmurray.com/640/360';

    beforeEach(() => {
      wrapper.setProps({
        image,
      });
    });

    it('has image', () => {
      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
      const card = wrapper.find(TCard).vm as unknown as ICard;
      expect(card.isImageShown).toBeTruthy();
      expect(wrapper.find('.t-card__image').isVisible()).toBe(true);
      expect(wrapper.find('figure.t-card__image-figure').isVisible()).toBe(true);
      expect(wrapper.find('img.t-card__image-element').isVisible()).toBe(true);
      expect(wrapper.find('img.t-card__image-element').attributes('src')).toBe(image);
    });

    it('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('set image scale on hover', () => {
    beforeEach(() => {
      wrapper.setProps({
        scaleImageOnHover: true,
      });
    });

    it('has class', () => {
      expect(wrapper.find('.t-card_has_image-scale-on-hover').isVisible()).toBe(true);
    });
    it('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
