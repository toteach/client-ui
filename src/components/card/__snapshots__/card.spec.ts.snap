// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TCard image renders correctly 1`] = `
"<article class=\\"t-card\\">
  <div class=\\"t-card__image\\">
    <figure class=\\"t-card__image-figure\\"><img src=\\"https://www.fillmurray.com/640/360\\" class=\\"t-card__image-element\\"></figure>
  </div>
  <!---->
  <!---->
  <!---->
</article>"
`;

exports[`TCard renders correctly 1`] = `
"<article class=\\"t-card\\">
  <!---->
  <!---->
  <!---->
  <!---->
</article>"
`;

exports[`TCard set image scale on hover renders correctly 1`] = `
"<article class=\\"t-card t-card_has_image-scale-on-hover\\">
  <!---->
  <!---->
  <!---->
  <!---->
</article>"
`;

exports[`TCard title renders correctly 1`] = `
"<article class=\\"t-card\\">
  <!---->
  <header class=\\"t-card__header\\">
    <h3 class=\\"t-card__title\\">
      Title
    </h3>
  </header>
  <!---->
  <!---->
</article>"
`;
